Hilla fails to start on Windows

<!--
Please READ these instructions & USE the issue template below, Thank You!

Only use GitHub issues for bugs and feature requests.   
For general support from the community, use Vaadin discord chat https://vaad.in/fusion instead.

NOTE: Issues concerning certain UI components should go to the components repository https://github.com/vaadin/web-components .

Good quality bug report increases the likelihood to get the bug fixed. A bad quality one will likely be just closed. Please use the following template to report bugs.
-->
### Description of the bug 
Hilla 2.0 fails to start on Windows.

### Minimal reproducible example
Generate a project `npx @hilla/cli init --react my-hilla-app` and run it (probably the issue is due to a path with spaces on Windows machine).

### Expected behavior
Project starts

### Actual behavior
Failure with the following error:
```
...
...
2023-03-09T13:32:52.569+02:00  INFO 18552 --- [onPool-worker-1] dev.hilla.parser.core.Parser             : Search for endpoints in directories [C:\Users\Tarek Oraby\Downloads\my-hilla-app\target\classes]
2023-03-09T13:32:52.704+02:00  INFO 18552 --- [onPool-worker-1] dev.hilla.parser.core.Parser             : JVM Parser finished successfully
'C:\Program' is not recognized as an internal or external command,
operable program or batch file.
2023-03-09T13:32:53.223+02:00 ERROR 18552 --- [io-8080-exec-10] c.v.flow.server.DefaultErrorHandler      : 

java.lang.IllegalStateException: com.vaadin.flow.server.ExecutionFailedException: Failed to run TypeScript endpoint generator
	at com.vaadin.base.devserver.AbstractDevServerRunner.getCause(AbstractDevServerRunner.java:758) ~[vaadin-dev-server-24.0.0.jar:na]
	at com.vaadin.base.devserver.AbstractDevServerRunner.getCause(AbstractDevServerRunner.java:754) ~[vaadin-dev-server-24.0.0.jar:na]
	at com.vaadin.base.devserver.AbstractDevServerRunner.handleRequest(AbstractDevServerRunner.java:617) ~[vaadin-dev-server-24.0.0.jar:na]
	at com.vaadin.flow.server.VaadinService.handleRequest(VaadinService.java:1534) ~[flow-server-24.0.0.jar:24.0.0]
	at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:399) ~[flow-server-24.0.0.jar:24.0.0]
	at com.vaadin.flow.spring.SpringServlet.service(SpringServlet.java:106) ~[vaadin-spring-24.0.0.jar:na]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:814) ~[tomcat-embed-core-10.1.5.jar:6.0]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:223) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:691) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:441) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:335) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:295) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.springframework.web.servlet.mvc.ServletForwardingController.handleRequestInternal(ServletForwardingController.java:141) ~[spring-webmvc-6.0.6.jar:6.0.6]
	at org.springframework.web.servlet.mvc.AbstractController.handleRequest(AbstractController.java:178) ~[spring-webmvc-6.0.6.jar:6.0.6]
	at org.springframework.web.servlet.mvc.SimpleControllerHandlerAdapter.handle(SimpleControllerHandlerAdapter.java:51) ~[spring-webmvc-6.0.6.jar:6.0.6]
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1081) ~[spring-webmvc-6.0.6.jar:6.0.6]
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:974) ~[spring-webmvc-6.0.6.jar:6.0.6]
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1011) ~[spring-webmvc-6.0.6.jar:6.0.6]
	at org.springframework.web.servlet.FrameworkServlet.doGet(FrameworkServlet.java:903) ~[spring-webmvc-6.0.6.jar:6.0.6]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:705) ~[tomcat-embed-core-10.1.5.jar:6.0]
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:885) ~[spring-webmvc-6.0.6.jar:6.0.6]
	at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:814) ~[tomcat-embed-core-10.1.5.jar:6.0]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:223) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53) ~[tomcat-embed-websocket-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100) ~[spring-web-6.0.6.jar:6.0.6]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.6.jar:6.0.6]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93) ~[spring-web-6.0.6.jar:6.0.6]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.6.jar:6.0.6]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201) ~[spring-web-6.0.6.jar:6.0.6]
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:116) ~[spring-web-6.0.6.jar:6.0.6]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:185) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:158) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:177) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:542) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:119) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:357) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:400) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:859) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1734) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[tomcat-embed-core-10.1.5.jar:10.1.5]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
Caused by: com.vaadin.flow.server.ExecutionFailedException: Failed to run TypeScript endpoint generator
	at dev.hilla.internal.TaskGenerateEndpointImpl.execute(TaskGenerateEndpointImpl.java:70) ~[engine-runtime-2.0.0.jar:na]
	at com.vaadin.flow.server.frontend.NodeTasks.execute(NodeTasks.java:301) ~[flow-server-24.0.0.jar:24.0.0]
	at com.vaadin.base.devserver.startup.DevModeInitializer.runNodeTasks(DevModeInitializer.java:372) ~[vaadin-dev-server-24.0.0.jar:na]
	at com.vaadin.base.devserver.startup.DevModeInitializer.lambda$initDevModeHandler$0(DevModeInitializer.java:313) ~[vaadin-dev-server-24.0.0.jar:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1804) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture$AsyncRun.exec(CompletableFuture.java:1796) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622) ~[na:na]
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165) ~[na:na]
Caused by: dev.hilla.engine.GeneratorException: Generator execution failed with exit code 1
	at dev.hilla.engine.GeneratorShellRunner.run(GeneratorShellRunner.java:66) ~[engine-core-2.0.0.jar:na]
	at dev.hilla.engine.GeneratorProcessor.process(GeneratorProcessor.java:46) ~[engine-core-2.0.0.jar:na]
	at dev.hilla.internal.TaskGenerateEndpointImpl.execute(TaskGenerateEndpointImpl.java:68) ~[engine-runtime-2.0.0.jar:na]
	... 10 common frames omitted

2023-03-09T13:32:53.234+02:00 ERROR 18552 --- [io-8080-exec-10] o.a.c.c.C.[.[.[/].[springServlet]        : Servlet.service() for servlet [springServlet] threw exception

dev.hilla.engine.GeneratorException: Generator execution failed with exit code 1
	at dev.hilla.engine.GeneratorShellRunner.run(GeneratorShellRunner.java:66) ~[engine-core-2.0.0.jar:na]
	at dev.hilla.engine.GeneratorProcessor.process(GeneratorProcessor.java:46) ~[engine-core-2.0.0.jar
...
...
```

### Versions:
    - Vaadin / Hilla version: 2.0
    - Node version: 18.13.0
    - Java version: OpenJDK Runtime Environment Corretto-18.0.0.37.1 (build 18+37-FR)
    - OS version: Windows 10
    - IDE/Editor (if applicable): IntelliJ IDEA

--- a/packages/java/engine-core/src/main/java/dev/hilla/engine/GeneratorShellRunner.java
+++ b/packages/java/engine-core/src/main/java/dev/hilla/engine/GeneratorShellRunner.java
@@ -8,6 +8,7 @@ import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
+import java.util.stream.Collectors;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -15,7 +16,6 @@ import org.slf4j.LoggerFactory;
 import com.vaadin.flow.server.frontend.FrontendUtils;
 
 final class GeneratorShellRunner {
-    private static final boolean IS_WINDOWS = FrontendUtils.isWindows();
     private static final Logger LOGGER = LoggerFactory
             .getLogger(GeneratorShellRunner.class);
     private static final Path TSGEN_PATH = Paths.get("node_modules", "@hilla",
@@ -25,12 +25,6 @@ final class GeneratorShellRunner {
 
     public GeneratorShellRunner(File rootDirectory, String nodeCommand) {
         this.rootDirectory = rootDirectory;
-
-        if (IS_WINDOWS) {
-            arguments.add("cmd.exe");
-            arguments.add("/c");
-        }
-
         arguments.add(nodeCommand);
         arguments.add(TSGEN_PATH.toString());
     }
@@ -41,14 +35,22 @@ final class GeneratorShellRunner {
 
     public void run(String input) throws InterruptedException, IOException {
         Objects.requireNonNull(input);
+        List<String> args;
+
+        if (FrontendUtils.isWindows()) {
+            args = List.of("cmd.exe", "/c", arguments.stream()
+                    .map(arg -> (arg.contains(" ") ? "\"" + arg + "\"" : arg))
+                    .collect(Collectors.joining(" ", "\"", "\"")));
+        } else {
+            args = arguments;
+        }
 
         if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("Executing command: {}", String.join(" ", arguments));
+            LOGGER.debug("Executing command: {}", String.join(" ", args));
         }
 
         var builder = new ProcessBuilder().directory(rootDirectory)
-                .command(arguments)
-                .redirectOutput(ProcessBuilder.Redirect.INHERIT)
+                .command(args).redirectOutput(ProcessBuilder.Redirect.INHERIT)
                 .redirectError(ProcessBuilder.Redirect.INHERIT);
 
         var process = builder.start();

